lib/syskit/models/deployment_group.rb

```diff
@@ -459,10 +464,11 @@ def use_deployment(
                         process_managers.default_run_options(model)
                     )
 
+                    read_only_deployment = read_only ? [deployment_name] : []
```

Comment:
Still wrong.

A deployment may have more than one task. Turns out that *if* you are using the default deployment `OroGen.project.Task => "name"` the two are the same, but it is not true in general.

`name_mappings` is complete AFAIK. That is, all values in `name_mappings` are all the tasks in the deployment. So, you can use `name_mappings.keys`

However, I would also let the caller pass specific names, i.e. you need to check whether `read_only` is an array before you set it to that.

Reply:
This is not that. A deployment can have multiple tasks, that's what you are not handling here. You are already dealing with the `use_` resolving multiple times...